src/shared/dropin.c

```diff
@@ -26,15 +26,16 @@
 int drop_in_file(const char *dir, const char *unit, unsigned level,
                  const char *name, char **ret_p, char **ret_q) {
 
-        char prefix[DECIMAL_STR_MAX(unsigned)];
+        char prefix[DECIMAL_STR_MAX(unsigned)] = {};
         _cleanup_free_ char *b = NULL, *p = NULL, *q = NULL;
 
         assert(unit);
         assert(name);
         assert(ret_p);
         assert(ret_q);
 
-        sprintf(prefix, "%u", level);
+        if (level != UINT_MAX)
+                sprintf(prefix, "%u", level);
```

Comment:
If you touch this, please switch to xsprintf

Reply:
Also, in `strjoin` below we unconditionally append `-` to prefix. Please move that into sprintf here.